<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
          "DTD/xhtml1-strict.dtd">
<html>
  

  <head>
    
    <title>numpy.core.numeric.AxisError</title>
    <meta name="generator" content="pydoctor 22.2.2.dev0"> 
        
    </meta>
    <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=0.75" />
    <link rel="stylesheet" type="text/css" href="bootstrap.min.css" />
    <link rel="stylesheet" type="text/css" href="apidocs.css" />
    <link rel="stylesheet" type="text/css" href="extra.css" />
</head>

  <body>

    <div id="banner">    
    <div>
        <a href="/">Home</a>
        &gt; numpy-1.22.3 <!-- This is a placeholder -->
        
        <!-- (<a href=""&gt;show all versions</a&gt;) -->
    </div>
</div>

    <nav class="navbar navbar-default">
  
  <div class="container">

    <div class="navbar-header">
      
      <div class="navlinks">
        <span class="navbar-brand">
          numpy <a href="index.html">API Documentation</a>
        </span>

        <a href="moduleIndex.html">
          Modules
        </a>

        <a href="classIndex.html">
          Classes
        </a>

        <a href="nameIndex.html">
          Names
        </a>

        <div id="search-box-container">
          <div class="input-group">
            <input id="search-box" type="search" name="search-query" placeholder="Search..." aria-label="Search" minlength="2" class="form-control" autocomplete="off" />
            
            <span class="input-group-btn">
              <a style="display: none;" class="btn btn-default" id="search-clear-button" title="Clear" onclick="clearSearch()"><img src="fonts/x-circle.svg" alt="Clear" /></a>
              <a class="btn btn-default" id="search-help-button" title="Help" onclick="toggleSearchHelpText()"><img src="fonts/info.svg" alt="Help" /></a>
            </span>
          </div>
        </div>

      </div>

      <div id="search-results-container" style="display: none;">
        
        <div id="search-buttons">
          <span class="label label-default" id="search-docstrings-button">
          <label class="checkbox-inline">
            <input type="checkbox" id="toggle-search-in-docstrings-checkbox" value="false" onclick="toggleSearchInDocstrings()">
              search in docstrings
            </input>
          </label>
          </span>
          
        </div>
        
        <noscript>
            <h1>Cannot search: JavaScript is not supported/enabled in your browser.</h1>
        </noscript>

        <div class="hint" id="search-help-box">
          <p class="rst-last">
      
            Search bar offers the following options:
            <ul>   
                <li>
                  <strong>Term presence.</strong> The below example searches for documents that 
                    must contain “foo”, might contain “bar” and must not contain “baz”: <code>+foo bar -baz</code>
                </li> 

                <li>
                  <strong>Wildcards.</strong> The below example searches for documents with words beginning with “foo”: <code>foo*</code>
                </li> 

                <li>
                  <strong>Search in specific fields.</strong> The following search matches all objects 
                  in "twisted.mail" that matches “search”: <code>+qname:twisted.mail.* +search</code>

                  <p>
                    Possible fields: 'name', 'qname' (fully qualified name), 'docstring', and 'kind'.
                    Last two fields are only applicable if "search in docstrings" is enabled.
                  </p>
                </li>

                <li>
                  <strong>Fuzzy matches.</strong> The following search matches all documents 
                  that have a word within 1 edit distance of “foo”: <code>foo~1</code>
                </li>
            </ul>

          </p>
        </div>

        <div id="search-status"> </div>

        <div class="warning" id="search-warn-box" style="display: none;">
          <p class="rst-last"><span id="search-warn"></span></p>
        </div>

        <table id="search-results">
          <!-- Filled dynamically by JS -->
        </table>
        
        <div style="margin-top: 8px;">
          <p>Results provided by <a href="https://lunrjs.com">Lunr.js</a></p>
        </div>
      </div>

    </div>
  </div>
</nav>

    

    <div class="container">

      <div class="page-header">
        <h1 class="class"><code><code><a href="index.html" class="internal-link">numpy</a></code><wbr></wbr>.<code><a href="numpy.core.html" class="internal-link" title="numpy.core">core</a></code><wbr></wbr>.<code><a href="numpy.core.numeric.html" class="internal-link" title="numpy.core.numeric">numeric</a></code><wbr></wbr>.<code><a href="numpy.core.numeric.AxisError.html" class="internal-link" title="numpy.core.numeric.AxisError">AxisError</a></code></code></h1>
        <div id="showPrivate">
          <button class="btn btn-link" onclick="togglePrivate()">Toggle Private API</button>
        </div>
      </div>

      <div class="categoryHeader">
        class documentation
      </div>

      <div class="extrasDocstring">
        <p><code><span class="py-keyword">class</span> <span class="py-defname">AxisError</span>(<a href="https://docs.python.org/3/library/exceptions.html#ValueError" class="intersphinx-link" title="ValueError">ValueError</a>, <a href="https://docs.python.org/3/library/exceptions.html#IndexError" class="intersphinx-link" title="IndexError">IndexError</a>): <a href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/numpy-1.22.3//numpy/core/_exceptions.py#L123" class="sourceLink">(source)</a></code></p>
        <p><a href="classIndex.html#numpy.core.numeric.AxisError">View In Hierarchy</a></p>
      </div>

      <div class="moduleDocstring">
        <div><p>Axis supplied was invalid.</p>
<p>This is raised whenever an <tt class="rst-docutils literal">axis</tt> parameter is specified that is larger
than the number of array dimensions.
For compatibility with code written against older numpy versions, which
raised a mixture of <code><a href="https://docs.python.org/3/library/exceptions.html#ValueError" class="intersphinx-link" title="ValueError">ValueError</a></code> and <code><a href="https://docs.python.org/3/library/exceptions.html#IndexError" class="intersphinx-link" title="IndexError">IndexError</a></code> for this situation, this
exception subclasses both to ensure that <tt class="rst-docutils literal">except ValueError</tt> and
<tt class="rst-docutils literal">except IndexError</tt> statements continue to catch <code><a href="https://pydocbrowser.github.io/numpy/latest/numpy.core.numeric.AxisError.html" class="intersphinx-link">AxisError</a></code>.</p>
<div class="rst-versionadded">
<span class="rst-versionmodified rst-added">New in version 1.13.</span></div>
<div class="rst-admonition rst-admonition-examples">
<p class="rst-first rst-admonition-title">Examples</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>array_1d = np.arange(10)
<span class="py-prompt">&gt;&gt;&gt; </span>np.cumsum(array_1d, axis=1)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">  ...</span>
<span class="py-except">numpy.AxisError: axis 1 is out of bounds for array of dimension 1</span>
</pre><p>Negative axes are preserved:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>np.cumsum(array_1d, axis=-2)
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">  ...</span>
<span class="py-except">numpy.AxisError: axis -2 is out of bounds for array of dimension 1</span>
</pre><p>The class constructor generally takes the axis and arrays'
dimensionality as arguments:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-builtin">print</span>(np.AxisError(2, 1, msg_prefix=<span class="py-string">'error'</span>))
<span class="py-output">error: axis 2 is out of bounds for array of dimension 1</span>
</pre><p>Alternatively, a custom exception message can be passed:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-builtin">print</span>(np.AxisError(<span class="py-string">'Custom error message'</span>))
<span class="py-output">Custom error message</span>
</pre></div>
<table class="fieldTable"><tr class="fieldStart"><td class="fieldName" colspan="2">Parameters</td></tr><tr><td class="fieldArgContainer"><span class="fieldArg">axis</span></td><td class="fieldArgDesc">The out of bounds axis or a custom exception message.
If an axis is provided, then <code><a href="https://pydocbrowser.github.io/numpy/latest/numpy.core.numeric.AxisError.html#ndim" class="intersphinx-link">ndim</a></code> should be specified as well.</td></tr><tr><td class="fieldArgContainer"><span class="fieldArg">ndim</span></td><td class="fieldArgDesc">The number of array dimensions.</td></tr><tr><td class="fieldArgContainer"><span class="fieldArg">msg​_prefix</span></td><td class="fieldArgDesc">A prefix for the exception message.</td></tr></table></div>
      </div>

      <div id="splitTables">
        <table class="children sortable" id="id342">
  
  
  <tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__init__" class="internal-link" title="numpy.core.numeric.AxisError.__init__">__init__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__str__" class="internal-link" title="numpy.core.numeric.AxisError.__str__">__str__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="classvariable">
    
    <td>Class Variable</td>
    <td><code><a href="#__slots__" class="internal-link" title="numpy.core.numeric.AxisError.__slots__">__slots__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#axis" class="internal-link" title="numpy.core.numeric.AxisError.axis">axis</a></code></td>
    <td>The out of bounds axis or <tt class="rst-docutils literal">None</tt> if a custom exception
message was provided. This should be the axis as passed by
the user, before any normalization to resolve negative indices.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#ndim" class="internal-link" title="numpy.core.numeric.AxisError.ndim">ndim</a></code></td>
    <td>The number of array dimensions or <tt class="rst-docutils literal">None</tt> if a custom exception
message was provided.</td>
  </tr><tr class="instancevariable private">
    
    <td>Instance Variable</td>
    <td><code><a href="#_msg" class="internal-link" title="numpy.core.numeric.AxisError._msg">_msg</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr>
</table>
        

          
      </div>

      <div id="childList">

        <div class="basemethod">
  
  
  <a name="numpy.core.numeric.AxisError.__init__">
    
  </a>
  <a name="__init__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__init__</span>(self, axis, ndim=<a href="https://docs.python.org/3/library/constants.html#None" class="intersphinx-link">None</a>, msg_prefix=<a href="https://docs.python.org/3/library/constants.html#None" class="intersphinx-link">None</a>):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/numpy-1.22.3//numpy/core/_exceptions.py#L191">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="numpy.core.numeric.AxisError.__str__">
    
  </a>
  <a name="__str__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__str__</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/numpy-1.22.3//numpy/core/_exceptions.py#L202">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="baseclassvariable">
  
  
  <a name="numpy.core.numeric.AxisError.__slots__">
    
  </a>
  <a name="__slots__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">__slots__</span>: <code><a href="https://docs.python.org/3/library/stdtypes.html#tuple" class="intersphinx-link">tuple</a>[<wbr></wbr><a href="https://docs.python.org/3/library/stdtypes.html#str" class="intersphinx-link">str</a>, <wbr></wbr><span class="rst-variable-ellipsis">...</span>]</code> =
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/numpy-1.22.3//numpy/core/_exceptions.py#L189">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="numpy.core.numeric.AxisError.axis">
    
  </a>
  <a name="axis">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">axis</span>: <code><a href="https://docs.python.org/3/library/functions.html#int" class="intersphinx-link">int</a></code>, <em>optional</em> =
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/numpy-1.22.3//numpy/core/_exceptions.py#L138">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    
    <div><p>The out of bounds axis or <tt class="rst-docutils literal">None</tt> if a custom exception
message was provided. This should be the axis as passed by
the user, before any normalization to resolve negative indices.</p>
<div class="rst-versionadded">
<span class="rst-versionmodified rst-added">New in version 1.22.</span></div>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="numpy.core.numeric.AxisError.ndim">
    
  </a>
  <a name="ndim">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">ndim</span>: <code><a href="https://docs.python.org/3/library/functions.html#int" class="intersphinx-link">int</a></code>, <em>optional</em> =
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/numpy-1.22.3//numpy/core/_exceptions.py#L140">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    
    <div><p>The number of array dimensions or <tt class="rst-docutils literal">None</tt> if a custom exception
message was provided.</p>
<div class="rst-versionadded">
<span class="rst-versionmodified rst-added">New in version 1.22.</span></div>
</div>
    
  </div>
</div><div class="baseinstancevariable private">
  
  
  <a name="numpy.core.numeric.AxisError._msg">
    
  </a>
  <a name="_msg">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">_msg</span> =
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/numpy-1.22.3//numpy/core/_exceptions.py#L194">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
    
  </div>
</div>

      </div>
    </div>

    <footer class="navbar navbar-default">

  
  <div class="container">
    <a href="index.html">API Documentation</a> for numpy,
  generated by <a href="https://github.com/twisted/pydoctor/">pydoctor</a>
    22.2.2.dev0 at 2022-03-08 17:29:58.
  </div>

  <!-- Search related scripts needs to be loaded at the end of HTML 
    parsing not to introduce overhead and display HTML data asap -->
  <script src="ajax.js" type="text/javascript"></script>
  <script src="searchlib.js" type="text/javascript"></script>
  <script src="search.js" type="text/javascript"></script>

</footer>

    <script src="pydoctor.js" type="text/javascript"></script>

  </body>
</html>